skimage/transform/_geometric.py

```diff
@@ -1352,8 +1352,8 @@ class SimilarityTransform(EuclideanTransform):
         Rotation angle, clockwise, as radians.
         Implemented only for 2D and 3D. For 3D, this is given in ZYX Euler
         angles.
-    translation : (dim,) array_like, optional
-        x, y[, z] translation parameters. Implemented only for 2D and 3D.
+    translation : (x, y[, z, ...]) sequence of float, length D, optional
+        Translation parameters for each axis. Implemented only for 2D and 3D.
```

Comment:
> It seems that D (as in "length D") is used to mean n, i.e., the image's number of dimensions. We write "an nD image" so n takes values 2, 3, ... (not D).

I don't mind updating this to "n" or "N". Though I do notice that translation is only supported for 2D and 3D. So the `...` should be removed as specifying more than xyz wouldn't work?

`params : (dim+1, dim+1)` (below) should also be updated according to whatever we agree on here.

Reply:
Oh, thanks @lagru.

We can't use `N` because this already denotes something else (number of points):
https://github.com/scikit-image/scikit-image/blob/d6ba6dba124675f1cf6fe0280c7dc15d87e4dd51/skimage/transform/_geometric.py#L42

Reply:
I find `D` (Dimensionality) to be much clearer than `n`, particularly when next to `N` (Number of points)

Reply:
@scott-vsi I hear you. Have you scanned the other modules to see which usage would currently dominate? Should we update "nD image" everywhere with "D-dim image?" In a sentence, "nD image" seems much more natural though...

Reply:
I agree, nD reads better, although it looks like n-D and N-D are also used pretty commonly (there are actually not that many uses of nD. I could see changing those to n-D or N-D).

skimage is a large package, so I doubt you could ever get complete consensus, but it looks like `D` dominates for the dimensionality of the image. It is used in [skimage._shared.utils.check_nD](https://github.com/scikit-image/scikit-image/blob/v0.21.0/skimage/_shared/utils.py#L568), [skimage.measure._moments.moments_coords](https://github.com/scikit-image/scikit-image/blob/v0.21.0/skimage/measure/_moments.py#L22), [skimage.measure._regionprops_utils.euler_number](https://github.com/scikit-image/scikit-image/blob/v0.21.0/skimage/measure/_regionprops_utils.py#L68), and [skimage.feature.template.match_template](https://github.com/scikit-image/scikit-image/blob/v0.21.0/skimage/feature/template.py#L47). And that is just from a quick spot check.